modules/memfault/memfault_ncs_metrics.c

```diff
@@ -13,6 +14,9 @@
 
 LOG_MODULE_REGISTER(memfault_ncs_metrics, CONFIG_MEMFAULT_NCS_LOG_LEVEL);
 
+static bool connected;
```

Comment:
These doesn't seem to be used if `CONFIG_MEMFAULT_NCS_LTE_METRICS` is disabled. They should probably be guarded to avoid build warnings.

Reply:
True, added guard